Update snipeit.sh for v7 Requirements

jerm opened the pull request:
*  update changelog

*  update snipeit DB user name to not collide with database name

*  remove Fedora surpport, add amazoni linux 2 and Alma 8/9

*  Add php8.2 support in centos/alma

*  Split apart mysql user creation and permissions grant. Can't do them
   as one operation anymore

*  In general, we'll only support LTS Ubuntu repos because of this, but if 
   they happen to ship with our preferred PHP versions and otherwise work w/o
   heroics, we'll include nominal support in the script

*  the Ondrej repo we use for installing arbitrary PHP versions doesn't
   (easily) support non-LTS versions of ubuntu, which means that 23.04 and 
   23.10 are problematic using the repo
   However, these versions ship with  php8.1 and php8.2 (respectively), so we
   don't even need the Ondrej repo for these, so we'll remove it

*  split-out ubuntu 23* versions
   since 2023 didn't produce and LTS ubuntu versions

*  remove remaining 8.2 references in Ubuntu 23* block, because they
   ship with new-enough php (for now.. see below)

*  update Ubuntu versions comment in code block


*  stop quoting run_as_app_user args
   because "$@" splits positional parameters, but quoting the args makes them
   one arg, which doesn't work

*  fix up ondrej repo

*  add new ubuntu/debian versions, change php to 8.2

Commit:
add new ubuntu/debian versions, change php to 8.2

Commit:
fix up ondrej repo

Commit:
Debian 12

Commit:
stop quoting run_as_app_user args

because "$@" splits positional parameters, but quoting the args makes
them one arg, which doesn't work

Commit:
split-out ubuntu 23* versions

since 2023 didn't produce and LTS ubuntu versions

the Ondrej repo we use for installing arbitrary PHP versions doesn't
(easily) support non-LTS versions of ubuntu, which means that 23.04 and
23.10 are problematic using the repo

However, these versions ship with  php8.1 and php8.2 (respectively), so
we don't even need the Ondrej repo for these, so we'll remove it

In general, we'll only support LTS Ubuntu repos because of this, but if
they happen to ship with our preferred PHP versions and otherwise work
w/o heroics, we'll include nominal support in the script

Commit:
update Ubuntu versions comment in code block

Commit:
remove remaining 8.2 references in Ubuntu 23* block

different vim config confusion made my regex misbehave :-/

Commit:
awsl2 support, php8.2 support in centos/alma, new mysql user creation code

Commit:
remove fedora surpport, add amazon and alma

Commit:
install php-sodium on al2. fix snipeit database name

Commit:
fix enterprise linux version in remi repo for v9 distros

Commit:
re-add missing progress bar to ensure correct command ordering

Commit:
update changelog

Commit:
replace some tabs with spaces. don't @ me

uberbrady commented on snipeit.sh:
```diff
@@ -228,11 +221,14 @@ install_composer () {
   log "git clone https://github.com/snipe/snipe-it $APP_PATH" & pid=$!
   progress
+  pushd $APP_PATH
+  git checkout jerm/snipeit-sh
```
Hrm, this something used for localdev that ought to be different for v7 normally?

jerm replied:
doh!  yes!  should just be main... or maybe master until we've completed the transition? Folk will get master once we switch to main, but will _not_ get main until we switch to master.  I know we're planning to do it immediately, but just in case we have to bail for some reason?

uberbrady commented on snipeit.sh:
```diff
@@ -228,11 +221,14 @@ install_composer () {
   create_user
   echo "* Creating MariaDB Database/User."
-  mysql -u root --execute="CREATE DATABASE snipeit;GRANT ALL PRIVILEGES ON snipeit.* TO snipeit@localhost IDENTIFIED BY '$mysqluserpw';"
+  mysql -u root --execute="CREATE DATABASE snipeit;CREATE USER snipeit_dbuser@localhost IDENTIFIED BY '$mysqluserpw'; GRANT ALL PRIVILEGES ON snipeit.* TO snipeit_dbuser@localhost;"
```
So weird that they pulled that 'classic' syntax out, but so be it. Progress marches on or something.

uberbrady commented on snipeit.sh:
```diff
@@ -916,8 +956,8 @@ echo "  ***Open http://$fqdn to login to Snipe-IT.***"
 echo "* Cleaning up..."
-rm -f snipeit.sh
-rm -f install.sh
+# rm -f snipeit.sh
```
This makes sense to me - snipeit.sh is part of the repo; no need for us to delete it, right?

uberbrady replied:
(and the same for install.sh)

jerm replied:
yeah. @snipe added that deletion back in the day. I pulled it because it was annoying to have to keep restoring it when testing, but I'm also not sure of the intent around deleting it.

Snipe, any input here?

snipe replied:
I can't imagine why I'd have done that - I didn't work much on this script at all (and largely loathe it's existence)

jerm replied:
good enough for me :-D thanks!

uberbrady reviewed (commented):
This looks excellent! Thank you so much for all of your very hard, detailed work on this - but I love the idea that we can install more 'cleanly' on Snipe-IT v7 going forward with all of this work. I did see one thing that had a 'jerm' in it that I wanted to double-check is actually an OK thing to have, and I would love it if it were possible to potentially re-integrate Fedora support, perhaps for a later V7.x version. But once we've cleared that up I'll be happy to integrate it all! Thanks again for your painstaking work on this - it couldn't have been easy, and is just so heavily detail-focused - I don't think I would've been able to do it. If you can let me know about those questions we'll get this merged in ASAP. Thank you!!!

Commit:
Oops! Don't checkout my custom branch

using master until we make the switch to main. As master will keep
working after we switch branches, main will _not_ work _until_ we switch, and if
for some reason we have to delay the branch migration, this will remain
broken until we fix it.

snipe commented on snipeit.sh:
```diff
@@ -228,11 +221,14 @@ install_composer () {
   log "git clone https://github.com/snipe/snipe-it $APP_PATH" & pid=$!
   progress
+  pushd $APP_PATH
+  git checkout master
```
This should stay `master`, and then switch to `main`, right?

jerm replied:
That's how I was thinking about it. Wait till we've officially landed on main before we switch, otherwise it's broken until then, and Gosh forbid we need to delay the switch for some reason

snipe replied:
That's what I was thinking too, just making sure we're on the same page. :)

Commit:
remove removal of installation scripts
